### DISCUSSION TITLE
UART bridge

### DISCUSSION THREAD

**Discussion by dudanov** [home-assistant/architecture#1138] [Entity Models]
Hi.

I suppose everyone knows that there are enough smart home devices that integrate with external cloud or local systems using so-called USB dongles or modules that somehow use `UART` as an interface for interacting with the end device. I even have a [project](https://github.com/dudanov/iot-uni-dongle) for such dongle.

I had an idea to implement a `UART` subsystem in the HA core similar to the existing `Bluetooth`. It will also be possible to create `ESPHome` devices that can easily provide a *secure and encrypted* bridge `HA device - UART device`, providing it through the core API to integration developers.

I don't have enough experience in the HA core architecture to implement it myself, but I want discussing this idea here.
It would be cool!

Thanks!

**Comment by frenck** [comment]
Hi there @dudanov 👋 

> I don't have enough experience in the HA core architecture to implement it myself, but I want discussing this idea here.

This is not the place for ideation or feature requests. For that purpose, we have the community forums. This repository is mainly for developers to propose and get approval on implementation details they want to implement.

../Frenck
